testnet/stacks-node/src/tests/signer/commands/block_commit.rs

```diff
@@ -0,0 +1,89 @@
+use super::context::{SignerTestContext, SignerTestState};
+use crate::tests::signer::v0::MultipleMinerTest;
+use madhouse::{Command, CommandWrapper};
+use proptest::prelude::{Just, Strategy};
+use std::sync::{Arc, Mutex};
+
+pub struct SubmitBlockCommitSecondaryMinerCommand {
+    miners: Arc<Mutex<MultipleMinerTest>>,
+}
+
+impl SubmitBlockCommitSecondaryMinerCommand {
+    pub fn new(miners: Arc<Mutex<MultipleMinerTest>>) -> Self {
+        Self { miners }
+    }
+}
+
+impl Command<SignerTestState, SignerTestContext> for SubmitBlockCommitSecondaryMinerCommand {
```

Comment:
Maybe reconsider the name `Command`? It works, but `Action` or similar might "feel" better for certain implementations.

Reply:
@cylewitruk, I'd open an issue in madhouse-rs to consider it separately.